### DISCUSSION TITLE
ObservableGauge hardcoded to sample once per minute?

### DISCUSSION THREAD

**Discussion by rurounijones** [open-telemetry/opentelemetry-dotnet#3422] [Q&A] [answered]
This may not be _strictly_ OpenTelemetry-DotNet related. I am not sure given that this and `System.Diagnostics.Metrics` can be entwined. But I'll ask just in case. I have created an `ObservableGauge` that I want to sample once per second. 

```csharp
_meter.CreateObservableGauge("simulation_frames_per_second_gauge",
() => { return GetSimulationFramesPerSecond(serverShortName); },
description: "The number of server simulation frames per second");
```
However I cannot find any documentation nor any APIs that indicate that it is possible to set frequency with which the Gauge is sampled and it appears to be hardcoded to once per minute only.

Am I missing something or is it not possible to set the frequency with which an `ObservableGauge` is sampled yet. If it _is_ possible and I am just missing it then please let me know if it is a global thing or if I can set it per `ObservableGauge`.

**Comment by cijothomas** [comment, marked as answer]
Observable instrument callbacks are performed along with [collection](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#collect). If the collection (or exporter) interval is 30 secs, then Observable instrument callbacks are performed every 30 secs. So depending on the exporter you use, you can control the export internal, and hence call back frequency. 

For pullbased (Prometheus), callbacks occur only when the scrape request comes.

(There is no ability to have different frequencies for Observable instrument callback invocation and Exporting. This was discussed in the spec, but was decided to be not included in initial version.)

**Reply by rurounijones** [reply]
For posterity this is the code I ended up with:

```csharp
.AddOtlpExporter((o, m) =>
{
    o.Protocol = OtlpExportProtocol.Grpc;
    o.HttpClientFactory = () =>
    {
        HttpClient client = new HttpClient();
        return client;
    };
    o.ExportProcessorType = OpenTelemetry.ExportProcessorType.Simple;
    m.PeriodicExportingMetricReaderOptions.ExportIntervalMilliseconds =
        (int) TimeSpan.FromSeconds(1).TotalMilliseconds;
})
```
